### PR TITLE
Add tests for EventEmitter, EnvExpressions, and Heartbeat

### PR DESCRIPTION
**Pull request by mythmon**
Also reduces logging, since it clutters up test runs.

r?

### CODE REVIEW

File: test/test-Heartbeat.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  });
+
+  let learnMoreEl = hb.notice.querySelector('.text-link');
+  learnMoreEl.click();
+
+  // triggers sending ping to normandy
+  closeAllNotifications();
+};
+
+//exports['test it opens an engagement page after interaction'] = assert => {
```
Comment (brittanystoroz) [inline test/test-Heartbeat.js]: Are you saving this for future reference? Is this particularly hard to test? Would something like this not work?

```javascript
  let hb = new Heartbeat(targetWindow, {
    testing: true,
    message: 'oh hai',
    engagementButtonLabel: 'Engage whooo',
    postAnswerURL: 'https://www.example.org/engaging',
  });

  tabs.on('open', (tab) => {
    assert.equal(tabs.activeTab.url, 'https://example.org/engaging');
    // Close engagement tab
    tabs[tabs.length - 1].close();
    done();
  });

  let engagementEl = hb.notice.querySelector('.notification-button');
  engagementEl.click();
```

Comment (mythmon) [inline test/test-Heartbeat.js]: I stubbed out all the tests as a first pass as a sort of to-do list. I wanted to get this up for review, and forgot that I left this one undone. Just a matter of timing.

I think your proposed test will work great. I'll add it to the patch and run it to make sure it works. Thanks!

Comment (mythmon) [inline test/test-Heartbeat.js]: I had to change `tabs.on('open', ...)` to `tabs.on('load', ...)`. On `open`, the tab had the url of `about:blank`. Changing that and adding `www.` to the assertion line made this work great.

**Comment by brittanystoroz** [comment]
r+

**Pull request merged**